### PR TITLE
Rails Prework Part 6: Make Helper code zeitwerk compliant

### PR DESCRIPTION
**Pull request by brendanshean**
## WHAT
1.  Remove `GraphQLController` reference to `current_user`
2. Clean up `NavigationHelper` and related code to pass in `current_user` as an argument.
3. Fix some constants assignments in `NavigationHelper` that used dynamic routes

## WHY
1.  It's redundant since `GraphQLController` inherits from `ApplicationController` which `includes QuillAuthentication`
2. This will make the code more transparent as to where `current_user` is coming from.  Also, specs were failing with zeitwerk checks failed since `current_user` wasn't implicitly available to the helpers
3. There are some constant assignments that used path helpers (e.g. `root_path` in `HOME_TAB = { name: HOME, url: routes.root_path }` that were relying on a routes variable which caused issues with zeitwerk.

## HOW
1.  Remove the private method
2. Update method interfaces to include `current_user` parameter
3. Move these into memoized variables.

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  (The answer should mostly be 'YES'. If you answer 'NO', please justify.)
Have you deployed to Staging? | (Possible answers: YES, Not yet - deploying now!, NO - non-app change, NO - tiny change)
Self-Review: Have you done an initial self-review of the code below on Github? |
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | (N/A or Yes)

### CODE REVIEW

File: services/QuillLMS/app/helpers/public_pages_helper.rb [hunk 1]
```diff
@@ -15,7 +15,8 @@ module PublicPagesHelper
 
   EVIDENCE_HANDBOOK_URL = "https://docsend.com/view/29vcfdqa5aupkmfp"
 
-  def should_render_react_component
+  def render_react_component?(current_user)
+    (current_user)
```
Comment (anathomical) [inline services/QuillLMS/app/helpers/public_pages_helper.rb:19]: I'm not clear on what this line is doing.

File: services/QuillLMS/app/lib/quill_authentication.rb [hunk 2]
```diff
@@ -9,7 +9,17 @@ module QuillAuthentication
...
+    helper_method(
+      :admin?,
+      :current_user,
+      :previewing_student_dashboard?,
+      :sign_out?,
+      :signed_in?,
+      :signed_in_outside_demo?,
+      :staff?,
+      :test_method,
+      :viewing_demo_account?
```
Comment (anathomical) [inline services/QuillLMS/app/lib/quill_authentication.rb:21]: Appreciate you alphabetizing this.

File: services/QuillLMS/app/views/teachers/shared/_scorebook_tabs.html.erb [hunk 3]
```diff
@@ -102,7 +102,12 @@
         </ul>
       </div>
     </div>
-    <%= render partial: 'pages/shared/mobile_sub_navbar', locals: { tabs: mobile_subnav_tabs, active_tab: determine_active_tab(current_path), active_subtab: determine_active_subtab(current_path) } %>
+
+    <%= render 'pages/shared/mobile_sub_navbar',
+        tabs: mobile_subnav_tabs(current_user),
+        active_tab: determine_active_tab(current_path),
+        active_subtab: determine_active_subtab(current_path)
```
Comment (anathomical) [inline services/QuillLMS/app/views/teachers/shared/_scorebook_tabs.html.erb:109]: Not a huge deal, but should we alphabetize this list of attributes as well?

**Review by anathomical — APPROVED** [review]

File: services/QuillLMS/app/helpers/public_pages_helper.rb (hunk 1 above)
Comment (brendanshean) [reply services/QuillLMS/app/helpers/public_pages_helper.rb:19]: Me either.  Good catch.

**Pull request merged**